src/utils/handle-errors.js

```diff
@@ -0,0 +1,57 @@
+import { UnhandledPromiseRejectionError, UncaughtExceptionError } from '../errors/test-run';
+
+const runningTests = {};
+let handlingTests  = false;
+
+function handleError (ErrorCtor, message) {
+    if (handlingTests) {
+        Object.values(runningTests).forEach(testRun => {
+            testRun.addError(new ErrorCtor(message));
+
+            removeRunningTest(testRun);
+        });
+    }
+    else {
+        /* eslint-disable no-process-exit */
+        /* eslint-disable no-console */
+        console.log(message);
+
+        setTimeout(() => process.exit(1), 0);
+        /* eslint-enable no-process-exit */
+        /* eslint-enable no-console */
+    }
+}
+
+function onUnhandledRejection (reason) {
+    if (reason.isRejectedDriverTask)
+        return;
+
+    const message = reason instanceof Error ? reason.stack : reason;
```

Comment:
`reason` can be a complex type. See unhanded promise rejection formatting (https://github.com/DevExpress/testcafe-hammerhead/blob/f3db95efd52754107b45da7ed86ef29eedcc0ff6/src/client/sandbox/node/window.js#L111) and related tests.